lib/src/test/java/io/ably/lib/test/common/Helpers.java

```diff
@@ -557,7 +557,7 @@ public synchronized void onSuccess(T result) {
 		}
 
 		@Override
-		public void onError(ErrorInfo error) {
+		public synchronized void onError(ErrorInfo error) {
```

Comment:
I have caught java.lang.IllegalMonitorStateException when notify() invoked at this place. I think is correct implementation and must be used at this place.